File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    set := make(map[int]struct{})
    maximum := 0

    for _, num := range nums {
        set[num] = struct{}{}
    }

    for key, _ := range set {
        if _, contains := set[key - 1]; !contains {
            current := key + 1 
            longest := 1
            for {
                if _, contains := set[current]; contains {
                    current++
                    longest++
                } else {
                    break
                }
            }
            maximum = max(maximum, longest)
        }
    }

    return maximum
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hash_set = set(nums)
        maximum = 0
        for num in hash_set:
            if not num-1 in hash_set:
                longest = 1
                current = num
                while current + 1 in hash_set:
                    longest += 1
                    current += 1
                maximum = max(longest, maximum)

        return maximum